docs/usage/key-concepts/how-it-works.md

```diff
@@ -0,0 +1,152 @@
+---
+title: How it works ?
+description: Learn all about Renovate's workflow
+---
+
+# Introduction
+
+Renovate works by searching for all the dependencies in your project and then look for updates for each.
+
+To be compatible with the variety of dependency-naming and versioning conventions, Renovate has defined modules for each known convention, and you can define your own too.
+
+Please add comments to the [issue#25091](https://github.com/renovatebot/renovate/issues/25091) if you wish to see a part (better) shown in the graph below
+
+## Modules
+
+The three modules are: [manager](../modules/manager/index.md), [datasource](../modules/datasource/index.md) and [versioning](../modules/versioning.md).
+
+They are used sequentially:
+
+1. the manager module looks for files based on their name and extract dependencies from them, each dependency has a datasource
+2. the datasource module looks for the existing versions of the dependency
+3. the versioning module search for a valid version regarding the dependency's version
+
+For example:
+
+1. the `gitlabci` manager finds a dependency named `python:3.10-alpine` of datasource `docker`
+2. the `docker` datasource looks for versions and finds `[python:3.9,python:3.9-alpine,python:3.10,python:3.10-alpine,python:3.11,python:3.11-alpine]`
+3. the `docker` versioning takes `python:3.11-alpine` as it is compatible with `python:3.10-alpine`
```

Comment:
```suggestion
1. The `gitlabci` manager finds a dependency: `python:3.10-alpine` which has the  `docker` datasource
2. The `docker` datasource looks for versions and finds: `[python:3.9,python:3.9-alpine,python:3.10,python:3.10-alpine,python:3.11,python:3.11-alpine]`
3. The `docker` versioning returns `python:3.11-alpine`, because that version is compatible with `python:3.10-alpine`
```

I'm not sure how a `docker` versioning can "take a version". I think you mean returns a version?

Question for maintainers: can a versioning module _really_ know what's compatible or not? I think saying "as it is compatible, or because that version is compatible" may overstate the actual smarts in the versioning module?